source/plugins/ruby/extension_utils.rb

```diff
@@ -109,5 +108,35 @@ def getNamespaceFilteringModeForDataCollection
       end
       return namespaceFilteringMode
     end
+
+    def getDataCollectionProfile
+      begin
+        dataCollectionStreamProfile = "Default"
+        dataTypes = []
+        dataTypeToStreamIdMap = Extension.instance.get_stream_mapping()
+        if !dataTypeToStreamIdMap.nil? && !dataTypeToStreamIdMap.empty?
+          dataTypes = dataTypeToStreamIdMap.keys
+          if (dataTypes.length >= 12)
+            dataCollectionStreamProfile = "Default"
+          elsif ((dataTypes.length <= 3) && (dataTypes.include?("CONTAINER_LOG_BLOB") || dataTypes.include?("CONTAINERINSIGHTS_CONTAINERLOGV2")) && dataTypes.include?("KUBE_EVENTS_BLOB"))
+            dataCollectionStreamProfile = "LogsAndEvents"
+          elsif (dataTypes.length == 2) && dataTypes.include?("LINUX_PERF_BLOB") && dataTypes.include?("INSIGHTS_METRICS_BLOB")
+            dataCollectionStreamProfile = "Performance"
+          elsif (dataTypes.length == 2) && dataTypes.include?("INSIGHTS_METRICS_BLOB") && dataTypes.include?("KUBE_PV_INVENTORY_BLOB")
+            dataCollectionStreamProfile = "PV"
+          elsif (dataTypes.length == 8) && dataTypes.include?("INSIGHTS_METRICS_BLOB") && dataTypes.include?("KUBE_POD_INVENTORY_BLOB") &&

```

Comment:
how about ContainerImageInventory, ContainerServiceLog, Heartbeat, and syslog for syslog

Reply:
> how about ContainerImageInventory, ContainerServiceLog, Heartbeat, and syslog for syslog

1. We dont support  ContainerImageInventory & ContainerServiceLog
2. Heartbeat is not part of the container insights and rather its part of the MDSD
3. syslog is not part of the container insights extension and its configured outside of the container insights extension if you look at the ARM templates.